### PR TITLE
[Feature-3575][*] Force Task Success

### LINKED ISSUE #3726: [Feature][ui] Front-end of Force success
**Describe the feature**
Front-end of Force success

image

#3575

### PR DESCRIPTION
**Pull request by chengshiwen**
## *Tips*
- *Thanks very much for contributing to Apache DolphinScheduler.*
- *Please review https://dolphinscheduler.apache.org/en-us/community/index.html before opening a pull request.*

## What is the purpose of the pull request

**New Feature: Force Task Success #3726 #3575**
- Only add the ExecutionStatus `FORCED_SUCCESS`
- Remove the operation ExecuteType `resume_from_forced_success`

this closes #3726 #3575

## Brief change log

- Base on the previous pr for ui: #3861 #4160, but there are some conflicts and the pr has not been updated for a long time
- Base on the previous pr for api and master: #3576, but there are a lot of conflicts and the pr has not been updated for a long time
- Merge the pr #3861 #4160 #3576, and resolve all conflicts
- Refactor some unreasonable code in frontend and backend
- **Remove the operation resume_from_forced_success**
- Fix the bug in frontend and backend joint debugging test
- Fix some check style and unit test error
- Remove redundant code

## Verify this pull request

*(Please pick either of the following options)*

This pull request is already covered by existing tests, such as *(please describe tests)*.

(or)

This change added tests and can be verified as follows:

*(example:)*

  - *Added dolphinscheduler-api tests in TaskInstanceControllerTest.java and TaskInstanceServiceTest.java for end-to-end.*
  - *Added dolphinscheduler-dao tests in TaskInstanceMapperTest.java for end-to-end.*
  - *Manually verified the change by testing locally for frontend to backend.*

**Gif Verify**:
image

**Video Verify**: See [Video Verify Link](https://recordit.co/BleQOXTziO)

### CODE REVIEW

**Review by break60 — APPROVED** [review]
front end +1

**Review by CalvinKirs — APPROVED** [review]
+1

File: dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/master/runner/MasterExecThread.java [hunk 1]
```diff
@@ -414,7 +414,7 @@ private void initTaskQueue() {
             if (task.isConditionsTask() || DagHelper.haveConditionsAfterNode(task.getName(), dag)) {
                 continue;
             }
-            if (task.getState().typeIsFailure() && !task.taskCanRetry()) {
+            if (task.getState().typeIsFailure() && !task.taskCanRetry() && !task.isConditionsTask()) {
```
Comment (lenboo) [inline dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/master/runner/MasterExecThread.java:417]: already judged by the above if.

Comment (chengshiwen) [reply dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/master/runner/MasterExecThread.java:417]: Ok, I will revise it

**Comment by chengshiwen** [comment]
I will remove the operation `resume_from_forced_success` and revise all code logic @lenboo @lgcareer @CalvinKirs @break60

**Comment by dailidong** [comment]
> I will remove the operation `resume_from_forced_success` and revise all code logic @lenboo @lgcareer @CalvinKirs @break60

any progress？

**Comment by chengshiwen** [comment]
> > I will remove the operation `resume_from_forced_success` and revise all code logic @lenboo @lgcareer @CalvinKirs @break60
> 
> any progress？

Not yet since preparing marriage propose in the past few days. I will finish it today or tomorrow.

**Comment by chengshiwen** [comment]
The operation `resume_from_forced_success` has been removed. Please review @dailidong @CalvinKirs @lenboo @lgcareer @break60 

**Gif Verify**:
image

**Video Verify**: See [Video Verify Link](https://recordit.co/BleQOXTziO)

**Review by dailidong — APPROVED** [review]
+1
great job, and the video gif is very clear

**Pull request merged**